pkg/cli/registry/login/login.go

```diff
@@ -273,86 +274,30 @@ func (o *LoginOptions) Validate() error {
 	if o.Credentials.Empty() {
 		return fmt.Errorf("Unable to determine registry credentials, please specify a service account or log into the cluster.")
 	}
-	if len(o.ConfigFile) == 0 {
-		return fmt.Errorf("Specify a file to write credentials to with --to")
-	}
 	return nil
 }
 
 func (o *LoginOptions) Run() error {
 	if !o.SkipCheck {
 		ctx := apirequest.NewContext()
 		creds := registryclient.NewBasicCredentials()
-		url := &url.URL{Host: o.HostPort}
-		creds.Add(url, o.Credentials.Username, o.Credentials.Password)
+		hostPortURL := &url.URL{Host: o.HostPort}
+		creds.Add(hostPortURL, o.Credentials.Username, o.Credentials.Password)
 		insecureRT, err := rest.TransportFor(&rest.Config{TLSClientConfig: rest.TLSClientConfig{Insecure: true}, UserAgent: rest.DefaultKubernetesUserAgent()})
 		if err != nil {
 			return err
 		}
 		c := registryclient.NewContext(http.DefaultTransport, insecureRT).WithCredentials(creds)
-		if _, err := c.Repository(ctx, url, "does_not_exist", o.Insecure); err != nil {
+		if _, err := c.Repository(ctx, hostPortURL, "does_not_exist", o.Insecure); err != nil {
 			return fmt.Errorf("unable to check your credentials - pass --skip-check to bypass this error: %v", err)
 		}
 	}
 
-	filename := o.ConfigFile
-	var contents []byte
-	if filename != "-" {
-		data, err := ioutil.ReadFile(filename)
-		if err != nil && !os.IsNotExist(err) {
-			return err
-		}
-		contents = data
-	}
-	if len(contents) == 0 {
-		contents = []byte("{}")
-	}
-
-	cfg := make(map[string]interface{})
-	if err := json.Unmarshal(contents, &cfg); err != nil {
-		return fmt.Errorf("unable to parse existing credentials %s: %v", filename, err)
-	}
-
-	obj, ok := cfg["auths"]
-	if !ok {
-		obj = make(map[string]interface{})
-		cfg["auths"] = obj
-	}
-	auths, ok := obj.(map[string]interface{})
-	if !ok {
-		return fmt.Errorf("the specified config file %s does not appear to be in the correct Docker config.json format: have 'auths' key of type %T", filename, obj)
-	}
-	auths[o.HostPort] = map[string]interface{}{
-		"auth": o.Credentials.Auth,
-	}
-	contents, err := json.MarshalIndent(cfg, "", "  ")
-	if err != nil {
+	ctx := &imageTypes.SystemContext{AuthFilePath: o.ConfigFile}
+	if err := config.SetAuthentication(ctx, o.HostPort, o.Credentials.Username, o.Credentials.Password); err != nil {
```

Comment:
This makes the code so much easier :sweat_smile: we should've done that long time ago.